.github/workflows/pypi.yml

```diff
@@ -0,0 +1,53 @@
+# roughly following https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+
+name: PyPI Deployment
+
+on: [push]
+
+jobs:
+  deploy:
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository and submodules
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Install MPI
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y mpich libmpich-dev
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest pytest-cov pytest-mock
+        pip install coveralls
+        pip install build
+        pip install -r requirements.txt
+    - name: Build bingocpp
+      run: |
+        ./.build_bingocpp.sh
+    - name: Build PyPI package
+      run: |
+        python -m build
+    - name: Remove wheel (was built for testing purposes)
+      run: |
+        cd dist
+        rm *.whl
+        cd ..
+    - name: Deploy to TestPyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+    - name: Deploy to PyPI (only on tagged commits)
+      if: startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
```

Comment:
`PYPI_API_TOKEN` and `TEST_PYPI_API_TOKEN` will have to be added as GitHub secrets.